lib/io.js

```diff
@@ -357,9 +353,10 @@ DatumReader.prototype = {
         var oldOffset = decoder._input.offset;
         var schemaIndex = decoder.readLong();
 
-        if (schemaIndex < 0 || schemaIndex >= writersSchema.schemas.length) {
```

Comment:
Yeah, this makes sense - and it seems to be somewhat related to my other commend about `valueOf`